### PR TITLE
Added own implementation of MediaType and removed spring lib

### PR DESCRIPTION
**Pull request by smals-mavh**
Functional thing to review:

An exception is thrown when a mediatype has multiple slashes, so "application/this/will/not/work".
Are we too strict? Or is this okay?

### CODE REVIEW

File: core/src/test/java/io/github/belgif/rest/guide/validator/core/ApiFunctionsTest.java [hunk 1]
```diff
@@ -57,16 +58,16 @@ void isLowerCamelCase() {
     @Test
     void isCompatibleMediaType() {
```
Comment (pvdbosch) [inline core/src/test/java/io/github/belgif/rest/guide/validator/core/ApiFunctionsTest.java:59]: This duplicates the tests in MediaTypeTest too much. It's sufficient to test all variant values of MediaType.includes in MediaTypeTest, and only test the behavior when having a list of media types here.

File: core/src/test/java/io/github/belgif/rest/guide/validator/core/model/MediaTypeTest.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import io.github.belgif.rest.guide.validator.core.model.helper.MediaType;
+import io.github.belgif.rest.guide.validator.core.model.helper.MediaTypeException;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class MediaTypeTest {
+
+    @Test
+    void testIncludes() {
```
Comment (pvdbosch) [inline core/src/test/java/io/github/belgif/rest/guide/validator/core/model/MediaTypeTest.java:12]: Too many tests in one test method. Ideally, one test case = one test method so they're independently executed. But to avoid verbosity, you can limit to test methods per "var mediaType = ....".

File: core/src/test/java/io/github/belgif/rest/guide/validator/core/model/MediaTypeTest.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        var mediaType = new MediaType("application/json");
+        assertTrue(mediaType.includes(new MediaType("application/problem+json")));
+        assertTrue(mediaType.includes(new MediaType("application/json")));
+        assertFalse(mediaType.includes(new MediaType("application/problem+xml")));
+        assertFalse(mediaType.includes(new MediaType("*/*")));
+
+        mediaType = new MediaType("multipart/*");
+        assertTrue(mediaType.includes(new MediaType("multipart/*")));
+        assertTrue(mediaType.includes(new MediaType("multipart/form-data")));
+        assertTrue(mediaType.includes(new MediaType("multipart/chunked")));
```
Comment (pvdbosch) [inline core/src/test/java/io/github/belgif/rest/guide/validator/core/model/MediaTypeTest.java:22]: In this test and below "*/*" one: remove tests that validate same logic:
* e.g. if form-data and chuncked tests don't trigger different includes logic, it suffices to only test one of them 
* e.g. for "*/*" test: "application/*" ~ "multipart/*", "application/problem+json" ~ "application/json"

File: core/src/test/java/io/github/belgif/rest/guide/validator/core/model/MediaTypeTest.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        mediaType = new MediaType("*/*");
+        assertTrue(mediaType.includes(new MediaType("*/*")));
+        assertTrue(mediaType.includes(new MediaType("application/problem+json")));
+        assertTrue(mediaType.includes(new MediaType("application/json")));
+        assertTrue(mediaType.includes(new MediaType("application/problem+xml")));
+        assertTrue(mediaType.includes(new MediaType("multipart/*")));
+        assertTrue(mediaType.includes(new MediaType("multipart/form-data")));
+        assertTrue(mediaType.includes(new MediaType("multipart/chunked")));
+        assertTrue(mediaType.includes(new MediaType("application/*")));
+    }
```
Comment (pvdbosch) [inline core/src/test/java/io/github/belgif/rest/guide/validator/core/model/MediaTypeTest.java:33]: corner case still to be tested:

"application/problem+json" doesn't include "application/json" (only the other way around)

**Comment by pvdbosch** [comment]
It's okay that media types with multiple slashes are rejected, they're invalid.

File: core/src/test/java/io/github/belgif/rest/guide/validator/core/model/MediaTypeTest.java (hunk 3 above)
Comment (smals-mavh) [reply core/src/test/java/io/github/belgif/rest/guide/validator/core/model/MediaTypeTest.java:22]: Not completely sure what you mean by "/" test: But I think the tests are now modified to only test each logic once.

**Review by pvdbosch — APPROVED** [review]
meant `*/*` tests instead of `/` but markdown formatting changed my comment :)
